File: Examen5/MinMax/minmax.py
import numpy as np
import scipy.optimize as opt

def fun(x):
    return np.exp(x) - 2*x - 10


# Encontrar el mínimo de una función f(x) en un intervalo [a, b]
a, b = -2, 2
xmin = opt.fminbound(fun, a, b)
print(xmin)

# Calcular el máximo
def newfun(x):
    return -fun(x)

a, b = 0, 3
xmax = opt.fminbound(newfun, a, b)
print(xmax)

File: Examen5/MinMax/minmax.m
f1=@(x) exp(x)-2*x-10;

%Ejemplo3: 'fminbnd' = Calcular el mínimo de una función f en un intervalo [a,b]
a=-4;
b=-2;
xmin=fminbnd(f1,a,b)

%Ejercicio: Calcular el máximo de una funcion f en un intervalo [a,b]
a=1;
b=3;
newf1=@(x) -1*f1(x); %Reflejar f1 en el eje x
fmaxbnd=@(newf1,a,b) fminbnd(newf1,a,b);
xmax=fmaxbnd(newf1,a,b)
